pandas/core/internals/construction.py

```diff
@@ -244,7 +259,7 @@ def init_dict(data, index, columns, dtype=None):
         if missing.any() and not is_integer_dtype(dtype):
             if dtype is None or np.issubdtype(dtype, np.flexible):
                 # GH#1783
-                nan_dtype = object
+                nan_dtype = np.dtype(object)
```

Comment:
in this PR we have np.dtype(object), np.dtype("object") and  np.dtype(np.object_). 
we should have a preference and be consistent.
maybe np.dtype("O") to be consistent with the repr.

Reply:
let's try to address this separately

Reply:
sgtm